mbuild/box.py

```diff
@@ -2,151 +2,333 @@
-        return "Box(mins={}, maxs={}, angles={})".format(self.mins, self.maxs, self.angles)
+        (Lx, Ly, Lz, xy, xz, yz) = self.box_parameters
+        if self._precision is not None:
+            precision = self._precision
+            desc = (f"Box: Lx={Lx:.{precision}f}, "
+                    f"Ly={Ly:.{precision}f}, "
+                    f"Lz={Lz:.{precision}f}, "
+                    f"xy={xy:.{precision}f}, "
+                    f"xz={xz:.{precision}f}, "
+                    f"yz={yz:.{precision}f}")
+        else:
+            desc = (f"Box: Lx={Lx}, "
+                    f"Ly={Ly}, "
+                    f"Lz={Lz}, "
+                    f"xy={xy}, "
+                    f"xz={xz}, "
+                    f"yz={yz}")
+        return desc
 
-class _BoxArray(np.ndarray):
-    """Subclass of np.ndarry specifically for mb.Box
+    def _from_vecs_to_lengths_tilt_factors(self):
+        # vectors should already be aligned by _normalize_box
+        v1 = self._box_vectors[0, :]
+        v2 = self._box_vectors[1, :]
+        v3 = self._box_vectors[2, :]
 
-    This subclass is meant to be used internally to store Box attribute array.
-    This subclass is modified so that its __setitem__ method is rerouted to the
-    corresponding setter method.
+        Lx = np.linalg.norm(v1)
+        Ly = np.linalg.norm(v2)
+        Lz = np.linalg.norm(v3)
 
-    Parameters
-    ----------
-    array : array-like object
-        This can be tuple, list, or any array-like object that can be usually
-        passed to np.array
-    var : str
-        Corresponding Box's attributes like "maxs", "mins", "lengths", "angles"
-    box : mb.Box
-        This is the Box contains this attribute (one level up of this array)
+        v1_dot_v2 = np.dot(v1, v2)
+        v1_dot_v3 = np.dot(v1, v3)
+        v2_dot_v3 = np.dot(v2, v3)
+
+        xy = v1_dot_v2 / (Lx * Ly)
+        xz = v1_dot_v3 / (Lx * Lz)
+        yz = (v2_dot_v3 - ((v1_dot_v2/Lx) * (v1_dot_v3/Lx))) / (Ly*Lz)
+
+        return (Lx, Ly, Lz, xy, xz, yz)
+
+    def _get_angles(self):
+        """Calculate the angles between the vectors that define the box.
+
+        Calculates the angles alpha, beta, and gamma from the Box object
+        attribute box_vectors.
+
+        """
+
+        vector_magnitudes = np.linalg.norm(self.box_vectors, axis=1)
+
+        a_dot_b = np.dot(self._box_vectors[0], self._box_vectors[1])
+        b_dot_c = np.dot(self._box_vectors[1], self._box_vectors[2])
+        a_dot_c = np.dot(self._box_vectors[0], self._box_vectors[2])
+
+        alpha_raw = b_dot_c / (vector_magnitudes[1] * vector_magnitudes[2])
+        beta_raw = a_dot_c / (vector_magnitudes[0] * vector_magnitudes[2])
+        gamma_raw = a_dot_b / (vector_magnitudes[0] * vector_magnitudes[1])
+
+        alpha = np.rad2deg(np.arccos(np.clip(alpha_raw, -1.0, 1.0)))
+        beta = np.rad2deg(np.arccos(np.clip(beta_raw, -1.0, 1.0)))
+        gamma = np.rad2deg(np.arccos(np.clip(gamma_raw, -1.0, 1.0)))
```

Comment:
NumPy simplification
```suggestion
        (alpha, beta, gamma) = np.rad2deg(np.arccos(np.clip([alpha_raw, beta_raw, gamma_raw], -1.0, 1.0)))
```

Reply:
You learn something new every day. I did not know that numpy arrays (or vectors i guess in this case) can use the same multiple assignment syntax.